lib/functoria/app.ml

```diff
@@ -415,20 +415,16 @@ module Make (P : S) = struct
   let with_output i = function None -> i | Some o -> Info.with_output i o
 
   let configure_main ~argv i (init, jobs) : unit Action.t =
-    let main = match Info.output i with None -> "main" | Some f -> f in
-    let file = main ^ ".ml" in
-    Log.info (fun m -> m "Generating: %s" file);
-    Codegen.set_main_ml file;
-    Codegen.append_main "(* %s *)" (Codegen.generated_header ());
-    Codegen.newline_main ();
-    Codegen.append_main "%a" Fmt.text P.prelude;
-    Codegen.newline_main ();
-    Codegen.append_main "let _ = Printexc.record_backtrace true";
-    Codegen.newline_main ();
+    let main = Info.main i in
+    Log.info (fun m -> m "Generating: %a" Fpath.pp main);
+    Action.with_output ~path:main ~append:false ~purpose:"Create main.ml"
```

Comment:
shouldn't `purpose` use `main` then as well, instead of refering to `main.ml`?

Reply:
Thanks, fixed